ddtrace/propagation/http.py

```diff
@@ -423,6 +494,17 @@ def parent_call():
         if PROPAGATION_STYLE_B3_SINGLE_HEADER in config._propagation_style_inject:
             _B3SingleHeader._inject(span_context, headers)
 
+    @staticmethod
+    def _extract_header_value(possible_header_names, headers, default=None):
+        # type: (FrozenSet[str], Dict[str, str], Optional[str]) -> Optional[str]
+        for header in possible_header_names:
+            try:
```

Comment:
```suggestion
            if header in headers:
```

```python
In [8]: keys = ["asdf", "ASDF", "fdsa", "FDSA"]

In [9]: def fn_exc(d, default):
   ...:     for k in keys:
   ...:         try:
   ...:             return d[k]
   ...:         except KeyError:
   ...:             pass
   ...:     return default
   ...: 


In [10]: def fn_get(d, default):
    ...:     for k in keys:
    ...:         if k in d:
    ...:             return d[k]
    ...:     return default
    ...: 

In [11]: %timeit fn_exc({}, 123)
489 ns ± 2.35 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [12]: %timeit fn_get({}, 123)
130 ns ± 0.45 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [13]: %timeit fn_exc({"asdf": 321}, 123)
108 ns ± 0.524 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [14]: %timeit fn_get({"asdf": 321}, 123)
114 ns ± 0.175 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

some pretty big savings if we avoid the exception handling and there are many misses

Reply:
I will open a separate PR for this optimization.